shared/common-adapters/swipeable.native.tsx

```diff
@@ -27,34 +27,139 @@ const useActionsEnabled = (tx: Reanimated.SharedValue<number>) => {
   return {actionsEnabled}
 }
 
-const useSyncClosing = (
-  tx: Reanimated.SharedValue<number>,
-  swipeCloseRef?: React.MutableRefObject<(() => void) | null>
-) => {
-  const [hasSwiped, setHasSwiped] = React.useState(false)
-  const closeSelf = React.useCallback(() => {
+const useMakeCloseSelf = (swipeCloseRef?: React.MutableRefObject<(() => void) | null>) => {
+  return React.useCallback(() => {
     swipeCloseRef?.current?.()
     if (swipeCloseRef) {
       swipeCloseRef.current = null
     }
   }, [swipeCloseRef])
-  const closeOthersAndRegisterClose = React.useCallback(() => {
+}
+
+const makeSwipeClose = (
+  tx: Reanimated.SharedValue<number>,
+  swipeCloseRef: undefined | React.MutableRefObject<(() => void) | null>
+) => {
+  return function swipeClose() {
+    tx.value = Reanimated.withSpring(0, {
+      stiffness: 300,
+      damping: 30,
+    })
+    if (swipeCloseRef) swipeCloseRef.current = null
+  }
+}
+
+const useMakeCloseOthersAndRegisterClose = (
+  swipeCloseRef: undefined | React.MutableRefObject<(() => void) | null>,
+  setHasSwiped: (s: boolean) => void,
+  tx: Reanimated.SharedValue<number>
+) => {
+  return React.useCallback(() => {
     setHasSwiped(true)
     swipeCloseRef?.current?.()
     if (swipeCloseRef) {
-      swipeCloseRef.current = () => {
-        tx.value = Reanimated.withSpring(0, {
-          stiffness: 300,
-          damping: 30,
-        })
-        swipeCloseRef.current = null
-      }
+      swipeCloseRef.current = makeSwipeClose(tx, swipeCloseRef)
     }
   }, [swipeCloseRef])
+}
 
+const useSyncClosing = (
+  tx: Reanimated.SharedValue<number>,
+  swipeCloseRef?: React.MutableRefObject<(() => void) | null>
+) => {
+  const [hasSwiped, setHasSwiped] = React.useState(false)
+  const closeSelf = useMakeCloseSelf(swipeCloseRef)
+  const closeOthersAndRegisterClose = useMakeCloseOthersAndRegisterClose(swipeCloseRef, setHasSwiped, tx)
   return {closeSelf, closeOthersAndRegisterClose, hasSwiped}
 }
 
+const makePanOnStart = (
+  tx: Reanimated.SharedValue<number>,
+  startx: Reanimated.SharedValue<number>,
+  dx: Reanimated.SharedValue<number>,
+  started: Reanimated.SharedValue<boolean>,
+  closeOthersAndRegisterClose: () => void
+) => {
+  return function onStart() {
+    closeOthersAndRegisterClose()
```

Comment:
flipping how this works, doing the non worklet then worklet helps solve some memory leak problems trampolining between the various jsis here